### PR TITLE
[NEW] Implement first phase of the app request flow 

### PR DESCRIPTION
**Pull request by rique223**
## Jira task and ADR: [MKP-175](https://rocketchat.atlassian.net/browse/MKP-175?atlOrigin=eyJpIjoiYjU5NWIwOWM1MzkzNDdlZWE3ZWQzZmIxYWY4OTBjYTUiLCJwIjoiaiJ9)
I've brought back the logic used in the deprecated version of the request flow (the one that was canceled here: https://github.com/RocketChat/Rocket.Chat/pull/26772) since it will be a great quick start for phase 2 of the request flow (the more robust one with the use of rocket cat and emails). Also refactored and deprecated some endpoints to open them for end users.
Demo gif:
app_request_phase_1

## Steps to test or reproduce

1. Create a non-admin account;
2. Open marketplace;
3. Click the request option in the app menu of an app list app or the request button on the app details page.

The endpoints can be tested on postman or insomnia for the change of access;
## Further comments

[MKP-175]: https://rocketchat.atlassian.net/browse/MKP-175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AllanPazRibeiro — APPROVED** [review]
LGTM for the apps-engine part

**Review by juliajforesti — CHANGES_REQUESTED** [review]

File: apps/meteor/app/apps/client/orchestrator.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ class AppClientOrchestrator {
 	}
 
 	public async getApps(): Promise<App[]> {
-		const result = await APIClient.get<'/apps'>('/apps');
+		const result = await APIClient.get<'/apps/installed'>('/apps/installed');
```
Comment (juliajforesti) [inline apps/meteor/app/apps/client/orchestrator.ts:92]: IMO `getApps` should get **all** apps, not only installed ones - semantically speaking. Is it supposed to fetch just the installed ones? Maybe rename the method?

**Review by juliajforesti — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/marketplace/AppsList/AppRow.tsx [hunk 2]
```diff
@@ -105,7 +105,7 @@ const AppRow = (props: AppRowProps): ReactElement => {
 				)}
 				<AppStatus app={props} isAppDetailsPage={false} installed={installed} />
 				<Box minWidth='x32'>
-					<AppMenu app={props} isAppDetailsPage={false} mis='x4' />
+					<AppMenu app={props} isAppDetailsPage={false} installed={installed} mis='x4' />
```
[the remark this answers is not in the document]
Comment (rique223) [reply apps/meteor/client/views/marketplace/AppsList/AppRow.tsx:108]: Solved

File: apps/meteor/app/apps/client/orchestrator.ts (hunk 1 above)
Comment (rique223) [reply apps/meteor/app/apps/client/orchestrator.ts:92]: It doesn´t necessarily returns installed apps but the only purpose of the return of this endpoint/function is to become the list of installed apps. So, I'll change the title of the function. Solved.

**Pull request merged**